ci/deps/actions-38-minimum_versions.yaml

```diff
@@ -45,7 +46,7 @@ dependencies:
   - pytables=3.6.1
   - python-snappy=0.6.0
   - pyxlsb=1.0.8
-  - s3fs=2021.08.0
+  - s3fs=0.4.0
```

Comment:
This dependency was bumped in 1.5 to a newer version: https://pandas.pydata.org/pandas-docs/stable/whatsnew/v1.5.0.html#backwards-incompatible-api-changes

So we shouldn't be testing an older version

Reply:
This was causing build issues that's why I pinned it at 0.4.0 https://github.com/pandas-dev/pandas/pull/49261#issuecomment-1297496447.

What is the verdict on what version we want to put in here for `s3fs`?

Reply:
@JMBurley cc for visibility

Reply:
Thanks, @kostyafarber .  I'm not sure if pandas **can** be built successfully with s3fs>=0.4.0 (and other packages active that require `boto3`), and **I know that pandas cannot be packaged into larger production environments when it enforces** `s3fs>=0.4.0` because professional codebases need other `boto3` imports.

However, I'm not a pandas maintainer so I don't get to make an assertion on what to pin s3fs to.  @mroeschke  controls the initial take on this.  I'd make the pitch for not advancing the s3fs version pin, because I do not know of any benefits we need (could be a lack of awareness on my part) and I do know of major python ecosystem headaches from advancing the pin.

My team pins `s3fs<=0.4.0` versions internally for this exact reason.  Ditto my colleagues controlling data science at other companies.

Reply:
Hi @JMBurley - could you please clarify about

>  I'm not sure if pandas can be built successfully with s3fs>=0.4.0 

? The `environment.yml` file has a later version of `s3fs`, and building pandas with the `environment.yml` file works fine 

https://github.com/pandas-dev/pandas/blob/0b931173c842a3476646b627422cf943d15288f6/environment.yml#L48